drivers/comparator/comparator_nrf_comp.c

```diff
@@ -0,0 +1,783 @@
+#endif
+
+#ifndef COMP_EXTREFSEL_EXTREFSEL_AnalogReference6
+BUILD_ASSERT(SHIM_NRF_COMP_DT_INST_EXTREFSEL(0) != COMP_NRF_COMP_EXTREFSEL_AIN6);
+#endif
+
+#ifndef COMP_EXTREFSEL_EXTREFSEL_AnalogReference7
+BUILD_ASSERT(SHIM_NRF_COMP_DT_INST_EXTREFSEL(0) != COMP_NRF_COMP_EXTREFSEL_AIN7);
+#endif
+#endif
+
+#if SHIM_NRF_COMP_DT_INST_MAIN_MODE_IS_SE(0)
+#ifndef COMP_REFSEL_REFSEL_Int1V8
+BUILD_ASSERT(SHIM_NRF_COMP_DT_INST_REFSEL(0) != COMP_NRF_COMP_REFSEL_INT_1V8);
+#endif
+
+#ifndef COMP_REFSEL_REFSEL_Int2V4
+BUILD_ASSERT(SHIM_NRF_COMP_DT_INST_REFSEL(0) != COMP_NRF_COMP_REFSEL_INT_2V4);
+#endif
+
+#ifndef COMP_REFSEL_REFSEL_AVDDAO1V8
+BUILD_ASSERT(SHIM_NRF_COMP_DT_INST_REFSEL(0) != COMP_NRF_COMP_REFSEL_AVDDAO1V8);
+#endif
+
+#ifndef COMP_REFSEL_REFSEL_VDD
+BUILD_ASSERT(SHIM_NRF_COMP_DT_INST_REFSEL(0) != COMP_NRF_COMP_REFSEL_VDD);
+#endif
+#endif
+
+#if SHIM_NRF_COMP_DT_INST_MAIN_MODE_IS_DIFF(0)
+#ifndef COMP_HYST_HYST_Hyst40mV
+BUILD_ASSERT(SHIM_NRF_COMP_DT_INST_HYST(0) != COMP_NRF_COMP_HYST_40MV);
+#endif
+
+#ifndef COMP_HYST_HYST_Hyst50mV
+BUILD_ASSERT(SHIM_NRF_COMP_DT_INST_HYST(0) != COMP_NRF_COMP_HYST_50MV);
+#endif
+#endif
+
+#if SHIM_NRF_COMP_DT_INST_MAIN_MODE_IS_SE(0)
+static const struct comp_nrf_comp_se_config shim_nrf_comp_config0 = {
+	.psel = SHIM_NRF_COMP_DT_INST_PSEL(0),
+	.sp_mode = SHIM_NRF_COMP_DT_INST_SP_MODE(0),
+	.isource = SHIM_NRF_COMP_DT_INST_ISOURCE(0),
+#if SHIM_NRF_COMP_DT_INST_REFSEL_IS_AREF(0)
+	.extrefsel = SHIM_NRF_COMP_DT_INST_EXTREFSEL(0),
+#endif
+	.refsel = SHIM_NRF_COMP_DT_INST_REFSEL(0),
+	.th_down = SHIM_NRF_COMP_DT_INST_TH_DOWN(0),
+	.th_up = SHIM_NRF_COMP_DT_INST_TH_UP(0),
+};
+#else
+static const struct comp_nrf_comp_diff_config shim_nrf_comp_config0 = {
+	.psel = SHIM_NRF_COMP_DT_INST_PSEL(0),
+	.sp_mode = SHIM_NRF_COMP_DT_INST_SP_MODE(0),
+	.isource = SHIM_NRF_COMP_DT_INST_ISOURCE(0),
+	.extrefsel = SHIM_NRF_COMP_DT_INST_EXTREFSEL(0),
+	.hyst = SHIM_NRF_COMP_DT_INST_HYST(0),
+};
+#endif
+
+static struct shim_nrf_comp_data shim_nrf_comp_data0;
+
+#if CONFIG_PM_DEVICE
+static bool shim_nrf_comp_is_resumed(void)
+{
+	enum pm_device_state state;
+
+	(void)pm_device_state_get(DEVICE_DT_INST_GET(0), &state);
+	return state == PM_DEVICE_STATE_ACTIVE;
+}
+#else
+static bool shim_nrf_comp_is_resumed(void)
+{
+	return true;
+}
+#endif
+
+static void shim_nrf_comp_start(void)
+{
+	if (shim_nrf_comp_data0.started) {
+		return;
+	}
+
+	nrfx_comp_start(shim_nrf_comp_data0.event_mask, 0);
+	shim_nrf_comp_data0.started = true;
+}
+
+static void shim_nrf_comp_stop(void)
+{
+	if (!shim_nrf_comp_data0.started) {
+		return;
+	}
+
+	nrfx_comp_stop();
+	shim_nrf_comp_data0.started = false;
+}
+
+static int shim_nrf_comp_pm_callback(const struct device *dev, enum pm_device_action action)
+{
+	ARG_UNUSED(dev);
+
+	if (action == PM_DEVICE_ACTION_RESUME) {
+		shim_nrf_comp_start();
+	}
+
+#if CONFIG_PM_DEVICE
+	if (action == PM_DEVICE_ACTION_SUSPEND) {
+		shim_nrf_comp_stop();
+	}
+#endif
+
+	return 0;
```

Comment:
This callback should not return 0 for actions that it does not implement.

Reply:
This matches the current implementation of pm_device_runtime. I agree that pm_device_runtime should support -ENOTUP for these calls, but it expects 0 whether implemented or not :)

Reply:
After 740eba1341ab2456c97390ea71d4be5d9ff8bab7 got merged, it is no longer needed to pretend that `PM_DEVICE_ACTION_TURN_ON` is implemented.